cdap-common/src/main/java/io/cdap/cdap/common/operation/LongRunningOperation.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.common.operation;
+
+import com.google.common.util.concurrent.ListenableFuture;
+import io.cdap.cdap.proto.operationrun.OperationError;
+import io.cdap.cdap.proto.operationrun.OperationMeta;
+import java.lang.reflect.Type;
+import java.util.function.Consumer;
+
+/**
+ * LongRunningOperation represents a long-running asynchronous operation.
+ *
+ * @param <T> Type of operation request
+ */
+public interface LongRunningOperation<T> {
+
+  Type getRequestType();
```

Comment:
Please add javadoc for interface methods.

Reply:
done